libfwupdplugin/fu-udev-device.c

```diff
@@ -943,6 +943,32 @@ fu_udev_device_pwrite (FuUdevDevice *self, goffset port, guint8 data, GError **e
 #endif
 }
 
+/**
+ * fu_udev_device_get_parent_name
+ * @self: A #FuUdevDevice
+ *
+ * Returns the name of the direct ancestor of this device
+ *
+ * Returns: string or NULL if unset or invalid
+ *
+ * Since: 1.4.2
+ **/
+const gchar *
+fu_udev_device_get_parent_name (FuUdevDevice *self)
+{
+#ifdef HAVE_GUDEV
+	FuUdevDevicePrivate *priv = GET_PRIVATE (self);
+	g_autoptr(GUdevDevice) parent = NULL;
+
+	g_return_val_if_fail (FU_IS_UDEV_DEVICE (self), NULL);
+
+	parent = g_udev_device_get_parent (priv->udev_device);
+	return parent == NULL ? NULL :g_udev_device_get_name (parent);
```

Comment:
The lifecycle here looks fishy. The `transfer:none` name of the parent belongs to the parent, but we're unreffing it when this function returns. I think this needs to be `g_strdup`'d and a `gchar *` returned.

Reply:
fixed